packages/cdk/lib/stacks/context-stack.ts

```diff
@@ -1,31 +1,129 @@
-    const vpc = Vpc.fromLookup(this, "Vpc", { vpcId });
+    this.vpc = Vpc.fromLookup(this, "Vpc", { vpcId });
+
+    const { contextParameters } = props;
+    const { engineName } = contextParameters;
+
+    switch (engineName) {
+      case "cromwell":
+        this.renderCromwellStack(props);
+        break;
+      case "nextflow":
+        this.renderNextflowStack(props);
+        break;
+      default:
+        throw Error(`Engine '${engineName}' is not supported`);
+    }
+  }
+
+  private renderCromwellStack(props: ContextStackProps) {
+    const batchProps = this.getCromwellBatchProps(props);
+    const batchStack = this.renderBatchStack(batchProps);
+
+    let jobQueue;
+    if (props.contextParameters.requestSpotInstances) {
+      jobQueue = batchStack.batchSpot.jobQueue;
+    } else {
+      jobQueue = batchStack.batchOnDemand.jobQueue;
+    }
+
+    const commonEngineProps = this.getCommonEngineProps(props);
+    new CromwellEngineStack(this, "cromwell", {
+      jobQueue,
+      ...commonEngineProps,
+    }).outputToParent(this);
+  }
 
-    const batchStack = new BatchStack(this, "Batch", { vpc, contextParameters: props.contextParameters });
+  private renderNextflowStack(props: ContextStackProps) {
+    const batchProps = this.getNextflowBatchProps(props);
+    const batchStack = this.renderBatchStack(batchProps);
 
-    const commonProps: CromwellEngineStackProps | NextflowEngineStackProps = {
-      vpc,
+    let jobQueue, headQueue;
+    if (props.contextParameters.requestSpotInstances) {
+      jobQueue = batchStack.batchSpot.jobQueue;
+      headQueue = batchStack.batchOnDemand.jobQueue;
+    } else {
+      headQueue = jobQueue = batchStack.batchOnDemand.jobQueue;
+    }
+
+    const commonEngineProps = this.getCommonEngineProps(props);
+    new NextflowEngineStack(this, "nextflow", {
+      ...commonEngineProps,
+      jobQueue,
+      headQueue,
+    }).outputToParent(this);
+  }
+
+  private getCromwellBatchProps(props: ContextStackProps) {
+    const commonBatchProps = this.getCommonBatchProps(props);
+    if (props.contextParameters.requestSpotInstances) {
+      return {
+        ...commonBatchProps,
+        createSpotBatch: true,
+        createOnDemandBatch: false,
+      };
+    }
+
+    return {
+      ...commonBatchProps,
+      createSpotBatch: false,
+      createOnDemandBatch: true,
+    };
+  }
+
+  private getCommonBatchProps(props: ContextStackProps) {
+    const { contextParameters } = props;
+    return {
+      vpc: this.vpc,
+      contextParameters,
+    };
+  }
+
+  private getNextflowBatchProps(props: ContextStackProps) {
+    const commonBatchProps = this.getCommonBatchProps(props);
+    if (props.contextParameters.requestSpotInstances) {
+      return {
+        ...commonBatchProps,
+        createSpotBatch: true,
+        createOnDemandBatch: true,
+      };
+    }
+
+    return {
+      ...commonBatchProps,
+      createSpotBatch: false,
+      createOnDemandBatch: true,
+    };
+  }
```

Comment:
Could this not be simplified to;
```
return {
        ...commonBatchProps,
        createSpotBatch: props.contextParameters.requestSpotInstances,
        createOnDemandBatch: true,
      };
```

Reply:
I think here I'd agree that it's an overkill to expand that much